cmd/skopeo/tags.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/containers/image/v5/docker"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+
+	"io"
+)
+
+// tagListOutput is the output format of (skopeo tags), primarily so that we can format it with a simple json.MarshalIndent.
+type tagListOutput struct {
+	DockerRepository string   `json:",omitempty"`
+	Tags             []string `json:",omitempty"`
+}
+
+type tagsOptions struct {
+	global *globalOptions
+	image  *imageOptions
+}
+
+func tagsCmd(global *globalOptions) cli.Command {
+	sharedFlags, sharedOpts := sharedImageFlags()
+	imageFlags, imageOpts := imageFlags(global, sharedOpts, "", "")
+
+	opts := tagsOptions{
+		global: global,
+		image:  imageOpts,
+	}
+
+	return cli.Command{
+		Name:  "list-repository-tags",
```

Comment:
I have a preference to name the command `list-tags` and require the input to follow `transport:....` convention. Maybe we want to support something similar for other transports in the future. Embedding "repository" into the command name and making it docker-transport only would close that door in some way.

The code wouldn't need to change much as we can error out for other transports for now. But I am late to the game and may miss context.

@mtrmac WDYT?

Reply:
My worry is that it’s not certain that things can be generalized for all transports.

It’s already _a bit_ conceptually dubious with `docker://`, where `docker://foo` means “the `docker://foo:latest` _image_”, not “the `docker://foo` _repository_”.

In general, the whole idea of “repository” and “tag” does generalize _at all_ well — compare https://github.com/containers/skopeo/pull/716#issuecomment-531382253 (besides syntax problems) suggesting that the whole local storage is a single “repository” and the hostname/[ns/…/]repo:tag value is a “tag”: it’s not at all an unreasonable position to take, but it makes the concept of “tag” meaningless and rules out the possibility of any transport-agnostic consumer of the output of this command.

(Compare https://github.com/containers/skopeo/pull/682#discussion_r295892434 and https://github.com/containers/skopeo/pull/524#issuecomment-469884710 dealing with registry names, all of this ends up with several `docker://` syntaxes with different meanings and rules.)

My default heuristic in such cases is to intentionally make the command extremely specific and limited; if we ever generalize it, we can easily add a general command, and maintain the specific one as a trivial shim. Going the other way from a prematurely-general interface to a specific one, or changing the prematurely-general interface to be general in a different way, requires better ability to forecast the future.

---

But, overall, sure:
- `list-tags docker://…` and `list-docker-tags …` are equivalent in the information that is being provided
- The “`docker://repo` means `:latest`” quibble is pedantic but does not make a difference  in practice (apart from our ability to cleanly reject tagged/digested input, but this PR does not reject it anyway)
- `list-repository-tags …` is already an specifics-losing compromise that I’m not at all strongly attached to

As long as the input is clearly namespaced and therefore unambiguous (i.e. not `skopeo list heuristically-guess-what-I-mean`), I think any of the three options works, and we’ve taken 4 months to decide and it’s not going to get any easier with 4 more months.

@vrothberg So, to settle this once for all, `skopeo list-tags docker://…`?